pennylane/labs/dla/cartan_subalgebra.py

```diff
@@ -0,0 +1,267 @@
+        is_orthogonal (bool): Whether the basis elements are all orthogonal, both within
+            and between ``g``, ``k`` and ``m``.
+
+    Returns:
+        np.ndarray: Adjoint vectors for new ordered :math:`\mathfrak{g}` of dimension :math:`|\mathfrak{g}| \times |\mathfrak{g}|`, ordered as :math:`\mathfrak{g} = \mathfrak{k} \oplus \tilde{\mathfrak{m}} \oplus \mathfrak{h}`
+        np.ndarray: Adjoint vectors for :math:`\mathfrak{k}` of dimension :math:`|\mathfrak{k}| \times |\mathfrak{g}|`.
+        np.ndarray: Adjoint vectors for :math:`\tilde{\mathfrak{m}}` of dimension :math:`|\tilde{\mathfrak{m}}| \times |\mathfrak{g}|`.
+        np.ndarray: Adjoint vectors for :math:`\mathfrak{h}` of dimension :math:`|\mathfrak{h}| \times |\mathfrak{g}|`.
+        np.ndarray: The new :math:`|\mathfrak{g}| \times |\mathfrak{g}| \times |\mathfrak{g}|` dimensional adjoint representation of :math:`\mathfrak{g}_\text{new}`, according to the new basis.
+
+    **Example**
+
+    A quick example computing a Cartan subalgebra of :math:`\mathfrak{su}(4)` using the Cartan involution :func:`~even_odd_involution`.
+
+    >>> import pennylane as qml
+    >>> from pennylane.labs.dla import cartan_decomp, cartan_subalgebra, even_odd_involution
+    >>> g = list(qml.pauli.pauli_group(2)) # u(4)
+    >>> g = g[1:] # remove identity -> su(4)
+    >>> g = [op.pauli_rep for op in g] # optional; turn to PauliSentence for convenience
+    >>> k, m = cartan_decomp(g, even_odd_involution)
+    >>> g = k + m # re-order g to separate k and m
+    >>> adj = qml.structure_constants(g)
+    >>> newg, k, mtilde, h, new_adj = cartan_subalgebra(g, k, m, adj)
+    >>> newg == k + mtilde + h
+    True
+    >>> h
+    [(1+0j) * Z(0) @ Z(1), (-1+0j) * Y(0) @ Y(1), (1+0j) * X(0) @ X(1)]
+
+    We can confirm that these all commute with each other, as the CSA is Abelian (= all operators commute).
+
+    >>> from pennylane.labs.dla import check_all_commuting
+    >>> check_all_commuting(h)
+    True
+
+    We can opt-in to return what we call adjoint vectors of dimension :math:`|\mathfrak{g}|`, where each component corresponds to an entry in (the ordered) ``g``.
+    The adjoint vectors for the Cartan subalgebra are in ``np_h``.
+
+    >>> np_newg, np_k, np_mtilde, np_h, new_adj = cartan_subalgebra(g, k, m, adj, return_adjvec=True)
+
+    We can reconstruct an operator by computing :math:`\hat{O}_v = \sum_i v_i g_i` for an adjoint vector :math:`v` and :math:`g_i \in \mathfrak{g}`.
+
+    >>> v = np_h[0]
+    >>> op = sum(v_i * g_i for v_i, g_i in zip(v, g))
+    >>> op = qml.simplify(op)
+    >>> op
+    Z(0) @ Z(1)
+
+    For convenience, we provide a helper function :func:`~adjvec_to_op` for the collections of adjoint vectors in the returns.
+
+    >>> from pennylane.labs.dla import adjvec_to_op
+    >>> h = adjvec_to_op(np_h, g)
+    >>> h
+    [Z(0) @ Z(1), (-1+0j) * (Y(0) @ X(1)), (-1+0j) * (X(0) @ Y(1))]
+
+    .. details::
+        :title: Usage Details
```

Comment:
I admit that so far I skipped this section (it's much easier to review reading the rendered version). When the rendered version is available I'll come back to it 👍